### PR TITLE
Allow host to turn on or off expense types for collectives

### PR DESCRIPTION
**Pull request by danielsum**
Resolve [https://github.com/opencollective/opencollective/issues/5525](https://github.com/opencollective/opencollective/issues/5525)
Resolve https://github.com/opencollective/opencollective/issues/5395
Related API PR: https://github.com/opencollective/opencollective-api/pull/7771

# Description

Here are the PR related to this issue : https://github.com/opencollective/opencollective/issues/5525

As discussed, the options to override the expenseTypes setting at the collective level is on stand by (juste let me know if we need to add it).

# Screenshots

Host_-_Admin_-_Open_Collective

dad2_-_New_expense_-_Open_Collective

dad_-_New_expense_-_Open_Collective

---

*Edit by @Betree: fix issue link & add API PR*

### CODE REVIEW

File: components/expenses/ExpenseForm.js [hunk 1]
```diff
@@ -700,6 +702,9 @@ ExpenseFormBody.propTypes = {
       }),
       settings: PropTypes.shape({
         disableGrantsByDefault: PropTypes.bool,
+        allowGrantSubmission: PropTypes.bool,
+        allowInvoiceSubmission: PropTypes.bool,
+        allowReceiptSubmission: PropTypes.bool,
```
Comment (Betree) [inline components/expenses/ExpenseForm.js:707]: Invalid proptypes leftover?

File: lib/collective.lib.js [hunk 2]
```diff
@@ -240,9 +240,13 @@ export const isIndividualAccount = account => ['USER', 'INDIVIDUAL'].includes(ac
...
   const accountHasGrantFlag = get(account, 'settings.expenseTypes.hasGrant');
+  const hostHasGrantFlag = get(host, 'settings.expenseTypes.hasGrant');
   if (isBoolean(accountHasGrantFlag)) {
     // If the account's feature flag is set, use it as the main source of truth
     return accountHasGrantFlag;
+  } else if (isBoolean(hostHasGrantFlag)) {
+    // If the host has configured grants to be opt-in
+    return hostHasGrantFlag;
   } else if (get(host, 'settings.disableGrantsByDefault')) {
     // If the host has configured grants to be opt-in, the previous flag (`hasGrant`) **must** be set
```
Comment (Betree) [inline lib/collective.lib.js:251]: We'll need to review that part in the context of https://github.com/opencollective/opencollective/issues/5395, but no need for you to iterate on it @danielsum

**Review by Betree — COMMENTED** [review]
Linting issue & minor stuff, the code looks great overall! CI seems to indicate a regression on  E2E, but it seems to originate in the API changes.

@danielsum The `Resolve` in the PR description is wrong, can it be updated?

**Comment by Betree** [comment]
Added a picker in the settings modal

image

**Comment by Betree** [comment]
https://user-images.githubusercontent.com/1556356/187656588-b1f485e4-d5e4-4d91-be9e-ddba00e91136.mp4

File: components/edit-collective/sections/Policies.js [hunk 3]
```diff
@@ -132,6 +145,7 @@ const Policies = ({ collective, showOnlyExpensePolicy }) => {
   const collectiveContributionPolicy = get(collective, 'contributionPolicy', null);
   const collectiveExpensePolicy = get(collective, 'expensePolicy', null);
   const collectiveDisableExpenseSubmission = get(collective, 'settings.disablePublicExpenseSubmission', false);
+  const expenseTypes = get(collective, 'settings.expenseTypes') || DEFAULT_SUPPORTED_EXPENSE_TYPES;
```
Comment (kewitz) [inline components/edit-collective/sections/Policies.js:148]: For consistency, I suggest using the default argument of `get`.
I would also be OK with updating all the lines above with this notation.
```suggestion
  const expenseTypes = get(collective, 'settings.expenseTypes', DEFAULT_SUPPORTED_EXPENSE_TYPES);
```

**Review by kewitz — APPROVED** [review]
Looks good! Not sure if necessary to update package-lock tho.

File: components/edit-collective/sections/Policies.js (hunk 3 above)
Comment (Betree) [reply components/edit-collective/sections/Policies.js:148]: If `settings.expenseTypes` is `null`, the default `get` argument won't be returned. `||` (or we could also use `??` here) seems safer.

**Pull request merged**